src/Uno.UI/Controls/UIFontHelper.iOS.cs

```diff
@@ -64,7 +64,9 @@ private static nfloat GetScaledFontSize(nfloat size, nfloat? basePreferredSize)
 				return size;
 			}
 
-			return size * (basePreferredSize / DefaultUIFontPreferredBodyFontSize) ?? (float)1.0;
+			var scaledSize = size * (basePreferredSize / DefaultUIFontPreferredBodyFontSize) ?? (float)1.0;
```

Comment:
You can use `Math.Min` to cap the value to `FeatureConfiguration.Font.MaximumTextScaleFactor`.

That being said, `scaledSize` is not the value to cap, it's the result of `basePreferredSize / DefaultUIFontPreferredBodyFontSize`. Otherwise, if will cap all values provided to `GetScaledFontSize`, and that's going to be very small (a font size of 2.0 at most).